cfme/tests/infrastructure/test_webmks_console.py

```diff
@@ -0,0 +1,169 @@
+                            ['creds']].get('username')
+    console_vm_password = credentials[provider.data.templates.get('console_template')
+                            ['creds']].get('password')
+
+    vm_obj.open_console(console='VM Console', invokes_alert=True)
+    assert vm_obj.vm_console, 'VMConsole object should be created'
+    vm_console = vm_obj.vm_console
+
+    request.addfinalizer(vm_console.close_console_window)
+    request.addfinalizer(appliance.server.logout)
+
+    # Get the login screen image, and make sure it is a jpeg file:
+    screen = vm_console.get_screen(180)
+    assert imghdr.what('', screen) == 'jpeg'
+
+    assert vm_console.wait_for_text(text_to_find="login:", timeout=200),\
+        "VM Console didn't prompt for Login"
+
+    # Enter Username:
+    vm_console.send_keys(console_vm_username)
+
+    assert vm_console.wait_for_text(text_to_find="Password", timeout=200),\
+        "VM Console didn't prompt for Password"
+    # Enter Password:
+    vm_console.send_keys("{}\n".format(console_vm_password))
+
+    time.sleep(5)  # wait for login to complete
+
+    # This regex can find if there is a word 'login','password','incorrect' present in
+    # text, irrespective of its case
+    regex_for_login_password = re.compile(r'\blogin\b | \bpassword\b| \bincorrect\b',
+     flags=re.I | re.X)
+
+    def _validate_login():
+        """
+        Try to read what is on present on the last line in console.
+
+        If it is word 'login', enter username, if 'password' enter password, in order
+        to make the login successful
+        """
+        if vm_console.find_text_on_screen(text_to_find='login', current_line=True):
+            vm_console.send_keys(console_vm_username)
+
+        if vm_console.find_text_on_screen(text_to_find='Password', current_line=True):
+            vm_console.send_keys("{}\n".format(console_vm_password))
+        # if the login attempt failed for some reason (happens with RHOS-cirros),
+        # last line of the console will contain one of the following words:
+        # [login, password, incorrect]
+        # if so, regex_for_login_password will find it and result will not be []
+        # .split('\n')[-1] splits the console text on '\n' & picks last item of resulting list
+        result = regex_for_login_password.findall(vm_console.get_screen_text().split('\n')[-1])
+        return result == []
+
+    # if _validate_login() returns True, it means we did not find any of words
+    # [login, password, incorrect] on last line of console text, which implies login success
+    wait_for(func=_validate_login, timeout=300, delay=5)
+
+    logger.info("Wait to get the '$' prompt")
+
+    vm_console.wait_for_text(text_to_find=provider.data.templates.get('console_template')
+                    ['prompt_text'], timeout=200)
+
+    # create file on system
+    vm_console.send_keys("touch blather\n")
+    logger.info("Wait 5 sec for file creation")
+    time.sleep(5)  # Wait for file creation
```

Comment:
It is required, arbitrary sleeps are not immune to networking problems. The problem is without wait_for any sleep we put in is permanent and everytime we need to wait, we need to wait that long. Since boxes can be under load, or connections take a hwile to connect etc, I prefer the wait_for, I odn't think it's that much extra work, and meas we can retry every 10 seconds, whilst having a maximum timeout of 2minutes if we wanted. 99% of the time it'll be done within 10 seconds, but if it has to wait longer it's not big deal.